plugins.d/Lets_Encrypt/get_certificate.py

```diff
@@ -141,17 +206,77 @@ def run():
-            'certificates.\n\nDo you wish to continue?',
+            'DNS must be configured before obtaining certificates.'
+            ' Incorrectly configured DNS and excessive attempts could'
+            ' lead to being temporarily blocked from requesting'
+            ' certificates.\n\n'
+            "You can check for a valid 'A' DNS record for your domain via"
+            ' Google:\n    https://toolbox.googleapps.com/apps/dig/\n\n'
+            'Do you wish to continue?',
             autosize=True
         )
         if ret != 'ok':
             return
+        d_conf = initial_load_conf()
+        write_conf(d_conf)
+
+    elif challenge == 'dns-01':
+        dns_01.initial_setup()
+        l_conf_possible = glob(path.join(dns_01.LEXICON_CONF_DIR, 'lexicon_*.yml'))
+        if len(l_conf_possible) == 0:
+            conf = None
+        elif len(l_conf_possible) == 1:
+            conf = l_conf_possible[0]
+        elif len(l_conf_possible) >= 2:
+            console.msgbox('Error',
+                           "Multiple lexicon_*.yml conf files found in"
+                           f" {LEXICON_CONF_DIR}, please ensure there is only"
+                           " one",
+                           autosize=True)
+            return
+
+        if conf:
+            provider = path.basename(conf).split('_', 1)[1][:-4]
+        else:
+            providers, err = dns_01.get_providers()
+            if err:
+                console.msgbox('Error', err, autosize=True)
+                return
 
-    if challenge == 'dns-01':
-        config = dns_01.load_config()
+            ret, provider = console.menu(
+                    'DNS providers list',
+                    "Select DNS provider you'd like to use",
+                    providers)
+            if ret != 'ok':
+                return
+
+            if provider == 'auto' and not which('nslookup'):
+                ret = console.yesno(
+                    'nslookup tool is required to use dns-01 challenge with'
+                    ' auto provider.\n\n'
+                    'Do you wish to install it now?',
+                    autosize=True
+                    )
+                if ret != 'ok':
+                    return
+                return_code, message = dns_01.apt_install(['dnsutils'])
+                if returncode != 0:
+                    console.msgbox('Error', message, autosize=True)
+                    return
+            if not provider:
+                console.msgbox('Error', 'No provider selected', autosize=True)
+
+        d_conf = initial_load_conf(provider)
+        conf_file, config = dns_01.load_config(provider)
+        if len(config) > 12:
+            console.msgbox(
+                    'Error',
+                    "Config file too big - needs to be 12 lines or less",
+                    autosize=True)
+            return
+        elif len(config) < 12:
+            for i in range(12 - len(config)):
+                config.append('')
```

Comment:
super silly nitpick, but you can do this without a loop

`config.extend([""] * (12 - len(config)))`

Reply:
Nice! :grin: - I'll implement that.